crates/evm/src/evm/handler.rs

```diff
@@ -557,9 +558,13 @@ fn calc_diff_size<EXT, DB: Database>(
             let account = &state[addr];
 
             if let Some(code) = account.info.code.as_ref() {
-                // if code is eoa code
-                diff_size += CODE_KEY_SIZE;
-                diff_size += code.len();
+                // Don't charge for account code if it is already in DB.
+                let db_code = db.code_by_hash(account.info.code_hash)?;
```

Comment:
db is revm's internal cache. we have to check if !self.code.exists(code_hash, working_set)

Reply:
We don't have an access to self.code here

Reply:
wait sorry, I got confused with journaled state. this is correct